src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_rpc_plugin.cc

```diff
@@ -53,11 +56,37 @@ bool RCRPCPlugin::Init(
                                           hmi_capabilities,
                                           policy_handler,
                                           *resource_allocation_manager_,
-                                          *interior_data_cache_));
+                                          *interior_data_cache_,
+                                          *this));
+
   interior_data_cache_.reset(new InteriorDataCacheImpl());
+
+  const auto& settings = app_manager.get_settings();
+  const auto frequency = settings.get_interior_vehicle_data_frequency();
+  max_requests_count = frequency.first;
+
+  exit_timer_signal_ = std::promise<void>();
+
+  timer_thread = std::thread(
+      std::bind1st(std::mem_fun(&RCRPCPlugin::TimerForResetRequestCount), this),
+      frequency.second);
   return true;
 }
 
+RCRPCPlugin::~RCRPCPlugin() {
+  exit_timer_signal_.set_value();
+  timer_thread.join();
+}
+
+void RCRPCPlugin::TimerForResetRequestCount(const uint32_t time_frame) {
+  std::future<void> futureObj = exit_timer_signal_.get_future();
+  while (futureObj.wait_for(std::chrono::milliseconds(1)) ==
+         std::future_status::timeout) {
+    std::this_thread::sleep_for(std::chrono::seconds(time_frame));
+    amount_of_request_in_this_time_frame_ = 0;
```

Comment:
@ZhdanovP looks like this variable may be accessed from different threads. It should be synchronized and wrapped by getters/setters, or getter and increment method.

Reply:
@LuxoftAKutsan I have created this var like 'std::atomic<uint32_t>' for synchronization, the get/incr will be added.